pkg/build/builder/docker.go

```diff
@@ -413,6 +419,33 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV Dockerfile instruction right after the
+// FROM instruction of the given dockerfile to define the environment variables
+// from env and returns the new Dockerfile.
+func insertEnvAfterFrom(env []kapi.EnvVar, dockerfile []byte) []byte {
+	if len(env) == 0 {
+		// Just return a copy.
+		return append([]byte(nil), dockerfile...)
+	}
+	fromLine := dockerfileFromLineRegexp.Find(dockerfile)
+	// Construct a byte slice of the form " NAME1=VALUE1 NAME2=VALUE2 ...".
+	buildEnv := []byte{}
+	for _, env := range env {
+		buildEnv = append(buildEnv, ' ')
```

Comment:
```go
env := fmt.Sprintf(" %s=%q", env.Name, env.Value)
```

Make sure you quote the env.Value... I can put \" in the string that will then corrupt the Dockerfile.